src/shared/Eis.cpp

```diff
@@ -0,0 +1,286 @@
+#include "Eis.hpp"
+#include "../core/PortalManager.hpp"
+#include "src/helpers/Log.hpp"
+#include <libeis.h>
+#include <sys/poll.h>
+#include <thread>
+
+EmulatedInputServer::EmulatedInputServer(std::string socketName) {
+    Debug::log(LOG, "[EIS] init socket: {}", socketName);
+
+    const char* xdg = getenv("XDG_RUNTIME_DIR");
+    if (xdg)
+        socketPath = std::string(xdg) + "/" + socketName;
+
+    if (socketPath.empty()) {
+        Debug::log(ERR, "[EIS] Socket path is empty");
+        return;
+    }
+
+    client.handle   = NULL;
+    client.seat     = NULL;
+    client.pointer  = NULL;
+    client.keyboard = NULL;
+    eis             = eis_new(NULL);
+
+    if (eis_setup_backend_socket(eis, socketPath.c_str())) {
+        Debug::log(ERR, "[EIS] Cannot init eis socket on {}", socketPath);
+        return;
+    }
+    Debug::log(LOG, "[EIS] Listening on {}", socketPath);
+
+    stop = false;
+    std::thread thread(&EmulatedInputServer::listen, this);
```

Comment:
this is bad, get the fd and incorporate it into the event loop poll thread in the main portalmgr.cpp

Reply:
This was not straightforward to impl, because i am creating one socket per session. I implemented something with a vector and a map inside the portalmgr (`addFdToEventLoop`, `removeFdFromEventLoop`)